Documenting remove_column as a reversible migration method

andreynering opened the pull request.

andreynering committed:
Documenting remove_column as a reversible migration method [ci skip]

claudiob commented:
@andreynering I think your addition makes sense.

However, you wrote this as an example:

```ruby
remove_column :posts, :slug, :string, null: false, index: true
```
and running `rake db:rollback` with that specific example might case an exception:

```sh
StandardError: An error has occurred, this and all later migrations canceled:

SQLite3::SQLException: Cannot add a NOT NULL column with default value NULL: ALTER TABLE "posts" ADD "slug" varchar NOT NULL
ActiveRecord::StatementInvalid: SQLite3::SQLException: Cannot add a NOT NULL column with default value NULL: ALTER TABLE "posts" ADD "slug" varchar NOT NULL

SQLite3::SQLException: Cannot add a NOT NULL column with default value NULL
```

The solution is simple: either add a `default` options or remove the `null` option from your example. Then rebase and you're all set. Oh… and don't forget to use `[ci skip]` in your commit message, so the tests are not run, since this PR only affects documentation :ribbon:

andreynering commented:
@claudiob Thank you for reviewing it.

Added `default` option in example and rebased.

`[ci skip]` was in the last commit too, but in the second line.

kaspth commented on guides/source/active_record_migrations.md:
```diff
@@ -539,6 +539,14 @@ definitions:
 
+`remove_column` is reversible if you supply the column type as a third
+argument. The column options should be provided too, so Rails will recreate the
+column exactly how it was before while rolling-back:
```
How about this :smile:
"Provide the original column options too, otherwise Rails can't recreate the column exactly when rolling back."

kaspth commented on guides/source/active_record_migrations.md:
```diff
@@ -539,6 +539,14 @@ definitions:
 `change_table` is also reversible, as long as the block does not call `change`,
 `change_default` or `remove`.
 
+`remove_column` is reversible if you supply the column type as a third
```
as *the* third :wink:

andreynering replied:
@kaspth Thanks!

kaspth commented:
Aside from the comments, :+1:

kaspth commented:
:heart:
